src/pool/srv_pool.c

```diff
@@ -2975,6 +2980,23 @@ ds_pool_update_internal(uuid_t pool_uuid, struct pool_target_id_list *tgts,
 	}
 
 	updated = true;
+
+	/* Update svc->ps_pool to match the new pool map. */
+	rc = ds_pool_tgt_map_update(svc->ps_pool, map_buf, map_version);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to update pool map cache: %d\n",
+			DP_UUID(svc->ps_uuid), rc);
+		/*
+		 * We must resign to avoid handling future requests with a
+		 * stale pool map cache.
+		 */
+		rdb_resign(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term);
```

Comment:
resign as the pool leader? I thought you can refresh the pool map cache by the buffer got from the rdb?

Reply:
If we reach here, we have failed to refresh the pool map cache or update the local pool group. But we have successfully changed the pool map in the RDB, so we step down and reply 0.